docker/Dockerfile.tcf-dev

```diff
@@ -86,33 +86,34 @@ RUN wget https://download.01.org/intel-sgx/linux-2.3.1/ubuntu18.04/sgx_linux_x64
  && rm sgx_linux_x64_sdk_2.3.101.46683.bin \
  && echo ". /opt/intel/sgxsdk/environment" >> /etc/environment
 
-# ("Untrusted") OpenSSL
 WORKDIR /tmp
-RUN wget https://www.openssl.org/source/openssl-1.1.1d.tar.gz \
- && tar -zxvf openssl-1.1.1d.tar.gz \
- && cd openssl-1.1.1d/ \
+# py-solc 3.2.0 requires an older version of the solc compiler, 0.4.25
+RUN mkdir -p $HOME/.py-solc/solc-v0.4.25/bin \
+  && curl -LsS https://github.com/ethereum/solidity/releases/download/v0.4.25/solc-static-linux -o /root/.py-solc/solc-v0.4.25/bin/solc \
+  && chmod 0755 /root/.py-solc/solc-v0.4.25/bin/solc
+
+# ("Untrusted") OpenSSL
+RUN OPENSSL_VER=1.1.1d \
+ && wget https://www.openssl.org/source/openssl-$OPENSSL_VER.tar.gz \
+ && tar -zxf openssl-$OPENSSL_VER.tar.gz \
+ && cd openssl-$OPENSSL_VER/ \
  && ./config \
  && THREADS=8 \
  && make -j$THREADS \
  && make test \
  && make install -j$THREADS \
  && ldconfig \
  && cd .. \
- && rm -rf openssl-1.1.1d
-# Note: we do _not_ delete openssl-1.1.1d.tar.gz as we re-use it below ..
-
-# py-solc 3.2.0 requires an older version of the solc compiler, 0.4.25
-RUN mkdir -p $HOME/.py-solc/solc-v0.4.25/bin \
-  && curl -LsS https://github.com/ethereum/solidity/releases/download/v0.4.25/solc-static-linux -o /root/.py-solc/solc-v0.4.25/bin/solc \
-  && chmod 0755 /root/.py-solc/solc-v0.4.25/bin/solc
+ && rm -rf openssl-$OPENSSL_VER
+# Note: we do _not_ delete openssl-*.tar.gz as we re-use it below ..
 
 # ("trusted") SGX OpenSSL
 # Note: This will compile in HW or SIM mode depending on the availability of
 # /dev/isgx and /var/run/aesmd/aesm.socket
 WORKDIR /tmp
 RUN git clone https://github.com/intel/intel-sgx-ssl.git  \
  && . /opt/intel/sgxsdk/environment \
- && (cd intel-sgx-ssl/openssl_source; mv /tmp/openssl-1.1.1d.tar.gz . ) \
+ && (cd intel-sgx-ssl/openssl_source; mv /tmp/openssl-*.tar.gz . ) \
```

Comment:
can we use  `mv openssl-$OPENSSL_VER.tar.gz .`

Reply:
We can. I have to define OPENSSL_VER again because the previous value was lost with the previous `RUN` command.